picoquic/tls_api.c

```diff
@@ -89,9 +89,37 @@ static void picoquic_init_openssl()
         ENGINE_register_all_ciphers();
         ENGINE_register_all_digests();
 #endif
+    } else {
+        openssl_is_init++;
     }
 }
 
+static void picoquic_release_openssl()
+{
+
+#if 0
+    /* Invalidating the openssl release code for now, as trying multiple open/close in
+     * the same process causes a crash */
+    if (openssl_is_init > 0) {
```

Comment:
I would not do this. You are right that these are leaked resources, but they should be leaked only once per process.
If you really want to release this resources, we should use some sort of global destructor.

Reply:
OK. There is in fact code in OpenSSL that attempts to release the resource on exit, so this may not be needed at all. In any case, I removed the ref counting and the attempt at automated release, and just added a global destructor to be called before exit. I only use it in the test program. It does free 37K out of the remaining 140K on the heap. After that, it seems that we are mostly left with file buffers for stdio, internal allocations for some system libraries, and possibly traces of openssl still left.

Most of the memory tests were done on Windows, because Visual Studio is a much nicer UI than valgrind. But we still need to see the result of valgrind on Linux.